plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQuerySplitSource.java

```diff
@@ -225,15 +223,22 @@ private static List<String> getProjectedColumnNames(List<BigQueryColumnHandle> c
         return columns.stream().map(BigQueryColumnHandle::name).collect(toImmutableList());
     }
 
-    private List<BigQuerySplit> createEmptyProjection(ConnectorSession session, TableDefinition.Type tableType, TableId remoteTableId, Optional<String> filter)
+    private List<BigQuerySplit> createEmptyProjection(ConnectorSession session, TableDefinition.Type tableType, TableId remoteTableId, Optional<String> filter, OptionalLong limit)
     {
         if (!TABLE_TYPES.containsKey(tableType)) {
             throw new TrinoException(NOT_SUPPORTED, "Unsupported table type: " + tableType);
         }
 
         // Note that we cannot use row count from TableInfo because for writes via insertAll/streaming API the number is incorrect until the streaming buffer is flushed
         // (and there's no mechanism to trigger an on-demand flush). This can lead to incorrect results for queries with empty projections.
-        String sql = selectSql(remoteTableId, "COUNT(*)", filter);
+        String sql;
+        if (limit.isPresent()) {
+            String innerSql = selectSql(remoteTableId, "*", filter, limit);
+            sql = "SELECT COUNT(*) FROM (%s)".formatted(innerSql);
+        }
+        else {
+            sql = selectSql(remoteTableId, "COUNT(*)", filter, limit);
```

Comment:
Do we make pushdowns for empty projection? From test seems that we are not, do we need this code?

Reply:
For this case, we need this code change
```
SELECT COUNT(*) FROM (SELECT * FROM %s LIMIT 3);
```